libraries/stdlib/src/kotlin/util/Tuples.kt

```diff
@@ -48,6 +52,30 @@ public infix fun <A, B> A.to(that: B): Pair<A, B> = Pair(this, that)
  */
 public fun <T> Pair<T, T>.toList(): List<T> = listOf(first, second)
 
+/**
+ * Transforms the first component of a [Pair] by applying the given [transform]
+ * function.
+ *  @sample samples.misc.Tuples.pairMapFirst
+ */
+public fun <A, B, T> Pair<A, B>.mapFirst(transform: (A) -> T): Pair<T, B> {
```

Comment:
Why these functions are not `inline`?

Reply:
> Why these functions are not `inline`?

Only because I'm not sure when its suitable to inline functions in Kotlin. :-) 
They can of course be inlined if you think that's a good idea? Is there any best practices   documented regarding when to inline or not inline functions somewhere?